test/test_ownedobject.py

```diff
@@ -132,14 +132,14 @@ def test_set_document(self):
         doc = sbol3.Document()
         c = sbol3.Component("scratch", sbol3.SBO_DNA)
         doc.add(c)
-        lsc = sbol3.LocalSubComponent(sbol3.SBO_DNA)
+        lsc = sbol3.LocalSubComponent([sbol3.SBO_DNA])
```

Comment:
I see that the pattern for a lot of constructors has changed, now requiring a list argument where a singleton was sufficient before.  It's not clear to me why this change necessarily follows from the improved exception handling.  Why were singleton arguments allowed previously?  This seems like it might break a lot of existing code (though we are still in alpha...).

Reply:
The singleton construct resulted in the string argument being processed as an iterable, as is often the case in Python. Without the list, and without the change in this pull request to catch the string argument, here is what happened:

```python
>>> import sbol3
RDFLib Version: 5.0.0
>>> lsc = sbol3.LocalSubComponent(sbol3.SBO_DNA)
>>> lsc.types
['h', 't', 't', 'p', 's', ':', '/', '/', 'i', 'd', 'e', 'n', 't', 'i', 'f', 'i', 'e', 'r', 's', '.', 'o', 'r', 'g', '/', 'S', 'B', 'O', ':', '0', '0', '0', '0', '2', '5', '1']
```
The unit tests did not catch this because they did not check the value in `types`. With the error message added these examples failed with the new checking for `Iterable`.

Reply:
I can attest to writing a lot of accidental `['h', 't', 't', 'p', 's', ...` lists that I only discover the hard way later.